src/Lib/Element.php

```diff
@@ -14,185 +14,150 @@
 
     protected $elements = [];
 
+    protected $externalOptions = [];
+
+    protected $externalElements = [];
+
     /**
-     * Add an option to this element
-     * Options are generally string key/ mixed value pairs
-     *
-     * @param string $key
-     * @param mixed $value
+     * Return the class name of the element converted into snake_case
      *
-     * @return $this
+     * @return string
      */
-    final protected function addOption($key, $value)
+    public function getElementName()
     {
-        $this->options[$key] = $value;
-        return $this;
+        $classNameWithNamespace = get_class($this);
+        $classNameWithoutNamespace = substr($classNameWithNamespace, strrpos($classNameWithNamespace, '\\')+1);
+        preg_match_all('/[A-Z][a-z]+/', $classNameWithoutNamespace, $classNameParts);
```

Comment:
Having said that, if the namespacing is also wrong *and* the unit tests are rubbish *and* it passes code review *and* the integration tests don't build and send the object, it could be pretty embarrassing.